compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -632,8 +698,17 @@ static LogicalResult setRootConfig(
   if (isAArch64(*variantOp) && !isQuantized) {
     return setAArch64RootConfig(entryPointFn, contractionOp, flowTileSizes,
                                 workgroupTileSizes, vectorSize);
-  } else if (!disableMatmulPadPipeline &&
-             (isX86(*variantOp) || isRISCV(*variantOp))) {
+  }
+
+  if (disableMatmulPadPipeline) {
+    return setMatmulNoPadRootConfig(entryPointFn, contractionOp, flowTileSizes,
+                                    workgroupTileSizes, vectorSize);
+  }
+  if (isX86(*variantOp)) {
+    return setX86RootConfig(entryPointFn, contractionOp, flowTileSizes,
+                            workgroupTileSizes, vectorSize);
+  }
+  if (isRISCV(*variantOp)) {
```

Comment:
It's enabled by default because I observed that it improves all our benchmarks on x86. I'm not able to measure RISC-V perf on my side, so I break the condition here. It's might not mature enough to RISC-V.

Do you think that we should keep x86 pipeline and RISC-V pipeline using same strategy and configurations? I'm happy to apply the change to RISC-V as well. The difficulty to me is that I'm not able to measure RISC-V performance, or triage the issues if there are regressions.